subscriptions-api-openapi.yaml

```diff
@@ -131,120 +138,341 @@ components:
     SubscriptionRequest:
       properties:
         protocol:
-          type: string
-          enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
-          description: REQUIRED. Identifier of a delivery protocol.
-          example: "HTTP"
-        protocolsettings:
-          type: object
-          anyOf:
-            - $ref: "#/components/schemas/HTTPSettings"
-            - $ref: "#/components/schemas/MQTTSettings"
-            - $ref: "#/components/schemas/AMQPSettings"
-            - $ref: "#/components/schemas/NATSSettings"
-            - $ref: "#/components/schemas/ApacheKafkaSettings"
-          description: OPTIONAL. A set of settings specific to the selected delivery protocol provider.
+          $ref: "#/components/schemas/Protocol"
+        protocolSettings:
+          oneOf:
+            - $ref: "#/components/schemas/ProtocolSettings"
+            - $ref:  "#/components/schemas/AMQPSettings"
+            - $ref:  "#/components/schemas/ApacheKafkaSettings"
+            - $ref:  "#/components/schemas/HTTPSettings"
+            - $ref:  "#/components/schemas/MQTTSettings"
+            - $ref:  "#/components/schemas/NATSSettings"
         sink:
           type: string
           format: url
           description: REQUIRED. The address to which events shall be delivered using the selected protocol.
           example: "https://endpoint.example.com/webhook"
+        sinkCredentials:
+          oneOf:
```

Comment:
same here - "anyOf" ?

Reply:
Actually, is "sinkCredentials" part of the spec? I can't find it. If we add this should it be part of the protocolSettings?